### DISCUSSION TITLE
Why I cannot build  when fallback : true  in dynamic pre_render pages

### DISCUSSION THREAD

**Discussion by thanhtutzaw** [vercel/next.js#39744] [Help] [answered]
I was learning SSG and I found I cannot use next build when fallback is true in getStaticPaths .
But when fallback is blocking , I can use next build.

**Comment by icyJoseph** [comment, marked as answer]
Hi,

This is because when using `fallback: true` you need to account for the `fallback` page, https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-pages, this is equivalent to a loading state, where the static props are not present yet.

This part is important:

```tsx
function Post({ post }) {
  const router = useRouter()

  // If the page is not yet generated, this will be displayed
  // initially until getStaticProps() finishes running
  if (router.isFallback) {
   // IMPORTANT bit
    return <div>Loading...</div>
  }

  // Render post...
}
```

When you use blocking, the server "waits" until the static props are ready, meanwhile it shows a blank page to the user.
